Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2290,8 +2290,8 @@ extension BrowserViewController: TabManagerDelegate {
             topTabsDidChangeTab()
         }
 
-       /// If the selectedTab is showing an error page trigger a reload
-        if let url = selected?.url, let internalUrl = InternalURL(url), internalUrl.isErrorPage {
+       /// If the selectedTab is showing an error page or stuck on loading trigger a reload
+        if let url = selected?.url, (InternalURL(url)?.isErrorPage ?? false || selected?.webView?.isLoading ?? false) {
```

Comment:
thanks for the contribution once again @imsanchit. I'm a little worried about this change because we might get the `isLoading` state and it doesn't necessarily mean that the loading is stuck. And we could be returning and missing the call `updateInContentHomePanel` where we have a lot of logic. I tested the change and this is the behaviour I'm seeing it only partially fix the issue once the tab is restored.

https://github.com/mozilla-mobile/firefox-ios/assets/18538058/14a8af02-8010-45a9-84a9-6f1f0d30637a

To be able merge we will like to have higher confidence that this change doesn't break the rest of the tab loading experience and it will best if we could find a solution for the whole problem like finding the root cause as to why didFinish delegate is not called.

Reply:
Understood, it may go to repeating re-loading in a few scenarios. What I sense is probably we shouldn't load the webview until it's being attached to some view controller, since OS suspends the process after some time, below is some console warning of this scenario. 
```2023-10-20 12:29:34.490033+0530 Client[48451:11304821] [assertion] Error acquiring assertion: <Error Domain=RBSServiceErrorDomain Code=1 "target is not running or doesn't have entitlement com.apple.runningboard.assertions.webkit" UserInfo={NSLocalizedFailureReason=target is not running or doesn't have entitlement com.apple.runningboard.assertions.webkit}>
2023-10-20 12:29:34.490253+0530 Client[48451:11304821] [ProcessSuspension] 0x128005800 - ProcessAssertion: Failed to acquire RBS assertion 'ConnectionTerminationWatchdog' for process with PID=49466, error: Error Domain=RBSServiceErrorDomain Code=1 "target is not running or doesn't have entitlement com.apple.runningboard.assertions.webkit" UserInfo={NSLocalizedFailureReason=target is not running or doesn't have entitlement com.apple.runningboard.assertions.webkit}
2023-10-20 12:29:34.502769+0530 Client[48451:11304821] [assertion] Error acquiring assertion: <Error Domain=RBSAssertionErrorDomain Code=2 "Specified target process does not exist" UserInfo={NSLocalizedFailureReason=Specified target process does not exist}>
2023-10-20 12:29:34.502988+0530 Client[48451:11304821] [ProcessSuspension] 0x128005860 - ProcessAssertion: Failed to acquire RBS assertion 'WebProcess Suspended Assertion' for process with PID=49466, error: Error Domain=RBSAssertionErrorDomain Code=2 "Specified target process does not exist" UserInfo={NSLocalizedFailureReason=Specified target process does not exist}```